internal/clients/http/azure/create_vms.go

```diff
@@ -38,6 +38,11 @@ func (c *client) CreateVMs(ctx context.Context, vmParams clients.AzureInstancePa
 		}
 
 		vmDescriptions[i].IPv4 = *publicIP.Properties.IPAddress
+		// can be nil if IPv6 is default in the future
+		privateIPv4 := networkInterface.Properties.IPConfigurations[0].Properties.PrivateIPAddress
```

Comment:
> 		// can be nil if IPv6 is default in the future

If this happends, would it end up with only PrivateIPAddress being nil? Everything else here are pointers/accessing IPConfigartions array that might be empty. Is this all safe?:D

Reply:
We have a `ptr.FromOrEmpty` generic helper, can you use that instead?

Reply:
it is not safe for sure xD it's YOLO I'll try to improve at least the ones I'm touching.
But the structure should be pretty stable, Azure uses pointers everywhere, but it either fails, or the data are there, so it's "a bit safer then complete YOLO" xD

Reply:
Will use the `ptr.FromOrEmpty` :+1:

Reply:
Done, I'll try to fix-up other occurences when I touch other Azure code :)

Reply:
Yeah, I do believe that for Azure this would not be inconsistent.